cmake/Testing.cmake

```diff
@@ -0,0 +1,159 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2013-2022 The Foundry Visionmongers Ltd
+
+# Don't re-process this module if it's already included by a project.
+include_guard(GLOBAL)
+if (NOT ${CMAKE_CURRENT_SOURCE_DIR} STREQUAL ${PROJECT_SOURCE_DIR})
+    # Ensure this module is `include`d from the top-level CMakeLists.txt
+    # (first), since we rely on this file only being included once (to
+    # prevent multiple definitions, e.g. openassetio-python-venv) and
+    # need all variables (e.g. find_package targets) to be made
+    # available project-wide.
+    message(FATAL_ERROR "Testing.cmake must be included in the top-level CMakeLists.txt")
+endif ()
```

Comment:
CMake variable scopes are a bit weird.  A lot of variables, it turns out, are "directory"-scoped, meaning they're visible in the current `CMakeLists.txt` and any child `CMakeLists.txt` added with  `add_subdirectory`.

This means if a variable is declared (including, as I discovered, build targets) in the top-level `CMakeLists.txt`, then it'll be available throughout the project.

Whereas, e.g. if we allow `openassetio-core/tests` to `include` this file, then that'll work fine.  But then `openassetio-core-c/tests` wouldn't be able to "see" anything declared in this file, because it's a different directory scope.

Hope this makes sense. Is there some way we could tweak the comment to make this a bit clearer?

Reply:
Oh I see, it's more an include order issue.

Not sure what else you can do to the comment, it's just an annoying niggle.